eclair-core/src/test/scala/fr/acinq/eclair/channel/RestoreSpec.scala

```diff
@@ -0,0 +1,229 @@
+package fr.acinq.eclair.channel
+
+import akka.actor.ActorRef
+import akka.actor.typed.scaladsl.adapter.actorRefAdapter
+import akka.testkit
+import akka.testkit.{TestActor, TestFSMRef, TestProbe}
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin._
+import fr.acinq.eclair.TestConstants.{Alice, Bob}
+import fr.acinq.eclair.blockchain.bitcoind.ZmqWatcher.WatchFundingSpentTriggered
+import fr.acinq.eclair.channel.states.ChannelStateTestsBase
+import fr.acinq.eclair.channel.states.ChannelStateTestsHelperMethods.FakeTxPublisherFactory
+import fr.acinq.eclair.crypto.Generators
+import fr.acinq.eclair.crypto.keymanager.ChannelKeyManager
+import fr.acinq.eclair.payment.relay.Relayer.{RelayFees, RelayParams}
+import fr.acinq.eclair.router.Announcements
+import fr.acinq.eclair.transactions.Scripts
+import fr.acinq.eclair.transactions.Transactions.{ClaimP2WPKHOutputTx, DefaultCommitmentFormat, InputInfo, TxOwner}
+import fr.acinq.eclair.wire.protocol.{ChannelReestablish, CommitSig, Error, Init, RevokeAndAck}
+import fr.acinq.eclair.{TestKitBaseClass, _}
+import org.scalatest.Outcome
+import org.scalatest.funsuite.FixtureAnyFunSuiteLike
+
+import scala.concurrent.duration._
+
+class RestoreSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with ChannelStateTestsBase {
+
+  type FixtureParam = SetupFixture
+
+  override def withFixture(test: OneArgTest): Outcome = {
+    val setup = test.tags.contains("disable-offline-mismatch") match {
+      case false => init()
+      case true => init(nodeParamsA = Alice.nodeParams.copy(onChainFeeConf = Alice.nodeParams.onChainFeeConf.copy(closeOnOfflineMismatch = false)))
+    }
+    import setup._
+    within(30 seconds) {
+      reachNormal(setup)
+      awaitCond(alice.stateName == NORMAL)
+      awaitCond(bob.stateName == NORMAL)
```

Comment:
nit: these aren't needed, they're already part of `reachNormal`

Reply:
```suggestion
```